code/_core/obj/projectile/laser.dm

```diff
@@ -93,11 +93,7 @@
 			on_projectile_hit(current_loc)
 			return FALSE
 
-/obj/projectile/bullet/flamer/on_enter_tile(var/turf/old_loc,var/turf/new_loc) //stolen from cult.
-
+/obj/projectile/bullet/flamer/on_enter_tile(turf/old_loc, turf/new_loc)
 	. = ..()
-
-	var/obj/effect/temp/hazard/flamethrowerfire = locate() in new_loc
-
-	if(!flamethrowerfire)
-		new /obj/effect/temp/hazard/flamethrowerfire(new_loc,SECONDS_TO_DECISECONDS(30),owner) //time and hazard type not final
\ No newline at end of file
+	if(new_loc != get_turf(owner)) // this stops you setting fire to yourself, unless you walk into the flames...
```

Comment:
I feel like there is a better way to fix this. If flamethrower fire is getting created on the turf that the owner is on, then there is something wrong with something else.

Reply:
I'll give a poke around, but when the projectile spawns, it instantly sets fire to the tile it spawned on, and because shoot spawns bullets on YOUR tile, Fwoosh. I don't know if it is intended to trigger on_enter_tile on spawn, but my guess is it does.